writer.go

```diff
@@ -84,3 +85,13 @@ func (w *Writer) Close() error {
 func (w *Writer) Write(p []byte) (n int, err error) {
 	return w.writeChunk(p, operationProcess)
 }
+
+// Reset initializes writer for reuse.
+func (w *Writer) Reset(dst io.Writer) {
+	encoderInitState(w)
+	w.params.quality = w.options.Quality
```

Comment:
This code for initializing the parameters is duplicated now. It should probably be moved to a separate method, or else incorporated into `encoderInitState`.